### DISCUSSION TITLE
Any Python front ends work well with python-socketio

### DISCUSSION THREAD

**Discussion by jamesbraza** [miguelgrinberg/python-socketio#1271] [Q&A] [answered]
I was using [Streamlit](https://streamlit.io/) for my front end, but I am learning it doesn't really play well with multithreading (and it considers `socketio.Client` events to be on a thread).

Anyone have recommendation(s) on a Python front end that works well with `python-socketio`?

I am trying to avoid writing JavaScript or HTML

**Comment by miguelgrinberg** [comment, marked as answer]
I don't believe that there is. And in case this isn't clear, it is not my intention to make the `Client` or `AsyncClient` compatible with Python-in-the-browser solutions. The supported platforms for the client are CPython and PyPy.
